mantidproject/mantid
# SANSILL: Add headers to ASCII exports

**Description of work.**

This PR adds a new property to `SaveAscii` algorithm `LogList`, which if speciifed leads to creation of a header in the ascii file containing the requested sample logs values and their respective units. 

This new feature is used in `DrILL` interface to add more information to ASCII-exported reduced data. The requested sample logs are specified in instruments' IPFs, and for SANS instruments they are: 

facility, instrument_name, experiment_identifier, 'experiment_title, start_time, end_time, numor_list (numor or list of numors), processedAs, wavelength, sample_transmission (numor or list of numors), collimation.actual_position, L2. 

If a specific sample log is not found, its value is replaced with 'Not defined'.

The facility information and sample transmission numors were not previously present in the sample logs of reduced data. This PR adds this information in SANSILLReduction, as well as improves the information stored in 'numor_list' log in comparison with contents of 'run_number', which previously was the numor of the first file in the list, and currently contains all of the numors that were used to create the output along with relevant binary operations.  

**To test:**

1. Test the functionality of the new property by running this minimal example either from the workbench or mantidpython:
```
from mantid.simpleapi import *

ws = CreateSampleWorkspace()
SaveAscii(ws, Filename='ascii_test.csv', LogList=['run_title', 'run_start', 'non_existent'])
```
2. Verify that the file saved in your default save directory is indeed a CSV , and that:
a) The header contains 3 lines with the left column filled with provided labels and right column with the read data, the data should be consistent what is available in the created sample workspace.
b) Verify that the value of the requested `non_existent` sample log is `Not defined`.

3. Test the functionality using DrILL interface. Make sure that ILL/D11 system test data is in your data search directories.
4. From the workbench, open the DrILL interface and fill the SampleRuns cell, e.g. with the following string: '2888+2889'. 
5. Using Export setting window, enable only SaveAscii. Process selected row(s).
6. Check if a 'sample_1_1.txt' file has been created in your default save directory.
7. Verify that it contains the header with the following sample logs:

facility, instrument_name, experiment_identifier, experiment_title, start_time, end_time, numor_list (numor or list of numors), processedAs, wavelength, sample_transmission (numor or list of numors), collimation.actual_position, L2.  



Fixes #30741 . 
Part of #30343 .





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
